shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -433,8 +447,9 @@ where
                 .await?
                 .into_items();
             // Go through the segment until we reach the offset.
-            // Unconditional progress is enum variant + compact encoding of number of elements
-            let mut progress = 1 + Compact::compact_len(&(items.len() as u64));
+            // Unconditional progress is enum variant, always 1 byte in SCALE encoding.
+            // (Segments do not have an item count, to make incremental writing easier.)
+            let mut progress = 1;
```

Comment:
Are we sure about the items count not being part of the count? I might have missed the pr mostly

Reply:
Archiver (in order to be able to encode incrementally) purposefully doesn't encode number of elements:
https://github.com/autonomys/subspace/blob/f5e8b43d13eb2d2e1072e3c3ac7d7a0f238d6ef4/crates/subspace-archiving/src/archiver.rs#L60-L70